### PR TITLE
[DONOTMERGE] Use pytest

### PR DESCRIPTION
**Pull request by elyezer**
This PR change from nose to py.test on Makefile in order to get better test run output.

It also drop Nose dependency which was around for a long time and was not being used.

Have done two commits just to separate things logically. Check its commit messages for more detailed info.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,30 +39,29 @@ test-docstrings:
...
 
 test-foreman-api:
-	$(NOSETESTS) -c robottelo.properties $(FOREMAN_API_TESTS_PATH)
+	$(PYTEST) $(PYTESTOPTS) $(FOREMAN_API_TESTS_PATH)
 
 test-foreman-api-threaded:
-	$(NOSETESTS) -c robottelo.properties $(FOREMAN_API_TESTS_PATH)\
-	    --processes=-1 --process-timeout=300
+	$(PYTEST) $(PYTESTOPTS) $(FOREMAN_API_TESTS_PATH)\
+		--processes=-1 --process-timeout=300
```
Comment (JacobCallahan) [inline Makefile]: does multi-threading work with pytest?

File: Makefile [hunk 2]
```diff
@@ -38,30 +39,29 @@ test-docstrings:
...
 
 test-foreman-cli:
-	$(NOSETESTS) -c robottelo.properties $(FOREMAN_CLI_TESTS_PATH)
+	$(PYTEST) $(PYTESTOPTS) $(FOREMAN_CLI_TESTS_PATH)
 
 test-foreman-cli-threaded:
-	$(NOSETESTS) -c robottelo.properties $(FOREMAN_CLI_TESTS_PATH)\
-	    --processes=-1 --process-timeout=300
+	$(PYTEST) $(PYTESTOPTS) $(FOREMAN_CLI_TESTS_PATH) --processes=-1 --process-timeout=300
```
Comment (JacobCallahan) [inline Makefile]: same question

File: Makefile (hunk 1 above)
Comment (elyezer) [inline Makefile]: Good question, will check

**Comment by omaciel** [comment]
I'd prefered to have seen maybe two different PRs:

* Use py.test instead of nose for Make file
* Remove nose all together and its calls

But ok, APT from me.

File: tests/foreman/cli/test_activationkey.py [hunk 3]
```diff
@@ -100,7 +99,6 @@ def _make_activation_key(self, options=None):
         {'name': gen_string('utf8', 15)},
         {'name': gen_string('html', 15)},
     )
-    @attr('cli', 'activation-key')
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_activationkey.py]: depending on which PR get's merged first, there may some merge conflict resolution needed. #1849

**Comment by elyezer** [comment]
Before

```
nosetests -c robottelo.properties tests/foreman/api/test_permission.py -mtest_search_permissions
/Users/elyezer/.virtualenvs/robottelo/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:734: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
F
======================================================================
FAIL: @test: search with no parameters return all permissions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/elyezer/code/robottelo/tests/foreman/api/test_permission.py", line 89, in test_search_permissions
    self.assertItemsEqual(PERMISSIONS_NAME, result_name)
AssertionError: Element counts were not equal:
First has 0, Second has 1:  u'view_registries'
First has 0, Second has 1:  u'create_registries'
First has 0, Second has 1:  u'destroy_registries'
First has 0, Second has 1:  u'search_repository_image_search'
First has 0, Second has 1:  u'view_discovery_rules'
First has 0, Second has 1:  u'new_discovery_rules'
First has 0, Second has 1:  u'edit_discovery_rules'
First has 0, Second has 1:  u'execute_discovery_rules'
First has 0, Second has 1:  u'delete_discovery_rules'
-------------------- >> begin captured stdout << ---------------------
[qe-foreman-rhel7.usersys.redhat.com] Executing task 'distro_info'
rhel 7 0

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Bugzilla bug 1184644 not in cache. Fetching.
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 5.984s

FAILED (failures=1)
```

After

```
 py.test --junit-xml=robottelo-results.xml tests/foreman/api/test_permission.py -ktest_search_permissions
=============================== test session starts ===============================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 266 items

tests/foreman/api/test_permission.py F

==================================== FAILURES =====================================
___________________ PermissionsTestCase.test_search_permissions ___________________

self = <tests.foreman.api.test_permission.PermissionsTestCase testMethod=test_search_permissions>

    @run_only_on('sat')
    def test_search_permissions(self):
        """@test: search with no parameters return all permissions

            @feature: Permission - Search

            @assert: Search returns a list of all expected permissions

            """
        result = entities.Permission().search()
        result_name = [permission['name'] for permission in result]
>       if bz_bug_is_open(1184644) and get_distro_info()[1] == 6:

tests/foreman/api/test_permission.py:87:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
robottelo/common/helpers.py:84: in get_distro_info
    return execute(distro_info, host=hostname)[hostname]
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/tasks.py:384: in execute
    multiprocessing
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/tasks.py:274: in _execute
    return task.run(*args, **kwargs)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/tasks.py:174: in run
    return self.wrapped(*args, **kwargs)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/automation_tools/__init__.py:1092: in distro_info
    release_info = run('cat /etc/redhat-release', quiet=True)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/network.py:647: in host_prompting_wrapper
    return func(*args, **kwargs)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/operations.py:1054: in run
    shell_escape=shell_escape)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/operations.py:923: in _run_command
    stderr=stderr, timeout=timeout)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/operations.py:807: in _execute
    worker.raise_if_needed()
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/thread_handling.py:12: in wrapper
    callable(*args, **kwargs)
../../.virtualenvs/robottelo/lib/python2.7/site-packages/fabric/io.py:231: in input_loop
    r, w, x = select([sys.stdin], [], [], 0.0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.capture.DontReadFromInput instance at 0x1093b6758>

    def fileno(self):
>       raise ValueError("redirected Stdin is pseudofile, has no fileno()")
E       ValueError: redirected Stdin is pseudofile, has no fileno()

../../.virtualenvs/robottelo/lib/python2.7/site-packages/_pytest/capture.py:441: ValueError
------------------------------ Captured stdout call -------------------------------
2015-01-22 14:46:33 - root - INFO - Bugzilla bug 1184644 not in cache. Fetching.
[qe-foreman-rhel7.usersys.redhat.com] Executing task 'distro_info'
------------------------------ Captured stderr call -------------------------------
/Users/elyezer/.virtualenvs/robottelo/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:734: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
----- generated xml file: /Users/elyezer/code/robottelo/robottelo-results.xml -----
=============== 265 tests deselected by '-ktest_search_permissions' ===============
==================== 1 failed, 265 deselected in 5.06 seconds =====================
```

PS.: py.test also add some color to the output.

**Comment by JacobCallahan** [comment]
APT and comments.

**Comment by elyezer** [comment]
Have discovered that Fabric does not like how py.test runs the tests. Will close this PR and will send a PR to drop nose dependency, and will work better on the test output later.

**Pull request closed without merging**